graphics/opengl/framebuffer.cpp

```diff
@@ -158,12 +158,23 @@ void FrameBuffer::init() {
 		glBindRenderbuffer(GL_RENDERBUFFER, 0);
 	} else {
 		glBindRenderbuffer(GL_RENDERBUFFER, _renderBuffers[0]);
-		glRenderbufferStorage(GL_RENDERBUFFER, useDepthComponent24() ? GL_DEPTH_COMPONENT24 : GL_DEPTH_COMPONENT16, _texWidth, _texHeight);
-		glFramebufferRenderbuffer(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_RENDERBUFFER, _renderBuffers[0]);
 
-		glBindRenderbuffer(GL_RENDERBUFFER, _renderBuffers[1]);
-		glRenderbufferStorage(GL_RENDERBUFFER, GL_STENCIL_INDEX8, _texWidth, _texHeight);
-		glFramebufferRenderbuffer(GL_FRAMEBUFFER, GL_STENCIL_ATTACHMENT, GL_RENDERBUFFER, _renderBuffers[1]);
+		const char *glVersion = (const char *)glGetString(GL_VERSION);
+		if (strstr(glVersion,"WebGL 1.0") != NULL) {
+			// See https://www.khronos.org/registry/webgl/specs/latest/1.0/#FBO_ATTACHMENTS
+			#define GL_DEPTH_STENCIL 0x84F9
```

Comment:
this should be guarded by `#ifnded` imo. other gl headers might define these constants (to the same value) - but you would get compiler warnings about redefining those, no?

Reply:
You are right. This is actually only needed because emscripten doesn't have all the WebGL headers (see emscripten-core/emscripten#4832 ). In theory this might be fixed in the future. I added `ifndefs` for this.